pandas/io/pytables.py

```diff
@@ -5066,7 +5066,7 @@ def _unconvert_string_array(
     if nan_rep is None:
         nan_rep = "nan"
 
-    data = libwriters.string_array_replace_from_nan_rep(data, nan_rep)
```

Comment:
wow i am shocked this didn't break anything (meaning how was this working before)

Reply:
the object returned here was the same as the input